src/cfn-mcp-server/awslabs/cfn_mcp_server/impl/tools/resource_scanner_tools.py

```diff
@@ -0,0 +1,267 @@
+                tag_key=tag_key,
+                tag_value=tag_value,
+                max_results=limit,
+                next_token=next_token,
+            )
+            resources = result['resources']
+        else:
+            # Get all resources if no filters
+            all_resources = cfn_utils.list_resource_scan_resources(latest_scan['ResourceScanId'])
+            resources = all_resources[:limit]
+            # If more than limit, set next_token (has more boolean)
+            result = {'next_token': None if len(all_resources) <= limit else 'has_more'}
+
+        # Format output
+        managed_resources = []
+        unmanaged_resources = []
+
+        for resource in resources:
+            resource_entry = {
+                'identifier': resource.get('ResourceIdentifier', {}),
+                'resource_type': resource.get('ResourceType'),
+            }
+
+            if resource.get('ManagedByStack'):
+                managed_resources.append(resource_entry)
+            else:
+                unmanaged_resources.append(resource_entry)
+
+        return {
+            'managed_resources': managed_resources,
+            'unmanaged_resources': unmanaged_resources,
+            'count': {
+                'managed': len(managed_resources),
+                'unmanaged': len(unmanaged_resources),
+                'total': len(resources),
+            },
+            'pagination': {
+                'limit': limit,
+                'next_token': result.get('next_token'),
+                'has_more': bool(result.get('next_token')),
+            },
+        }
+
+    except Exception as e:
+        raise handle_aws_api_error(e)
+
+
+async def list_related_resources_impl(
+    resources: List[Dict[str, Any]],
+    max_results: int = 100,
+    next_token: Optional[str] = None,
+    region: Optional[str] = None,
+) -> Dict[str, Any]:
+    """Simple implementation for listing AWS resources related to specified resources.
+
+    Args:
+        resources: List of resources to find related resources for, each resource should be a dictionary
+                   with 'resource_type' and 'resource_identifier' keys.
+                   Example: [{'resource_type': 'AWS::S3::Bucket', 'resource_identifier': 'my-bucket-123'}]
+
+        max_results: Maximum number of related resources to return (1-100)
+        next_token: AWS pagination token from previous response
+        region: AWS region to use
+
+    Returns:
+        Dict containing related resources and pagination info
+    """
+    # Basic validation
+    if not resources:
+        raise ClientError('Please provide at least one resource')
+
+    if len(resources) > 100:
+        raise ClientError('Maximum of 100 resources allowed')
+
+    if max_results < 1 or max_results > 100:
+        raise ClientError('max_results must be between 1 and 100')
+
+    try:
+        cfn_utils = CloudFormationUtils(region=region or 'us-east-1')
+
+        # Get latest completed scan
```

Comment:
function should accept an optional scanId and only do this if one is not provided